packages/cloudflare/src/workflows.ts

```diff
@@ -0,0 +1,207 @@
+                [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'task',
+              },
+            },
+            async span => {
+              try {
+                const result = await configOrCallback();
+                span.setStatus({ code: 1 });
+                return result;
+              } catch (error) {
+                captureException(error, { mechanism: { handled: true, type: 'cloudflare' } });
+                throw error;
+              } finally {
+                this._ctx.waitUntil(flush(2000));
+              }
+            },
+          );
+        });
+      });
+    } else if (typeof maybeCallback === 'function') {
+      // do(name, config, callback)
+      return this._step.do(name, configOrCallback, async () => {
+        // eslint-disable-next-line no-return-await
+        return await workflowStepWithSentry(this._instanceId, this._options, async () => {
+          // eslint-disable-next-line no-return-await
+          return await startSpan(
+            {
+              op: 'function.step.do',
+              name,
+              attributes: {
+                'cloudflare.workflow.timeout': configOrCallback?.timeout,
+                'cloudflare.workflow.retries.backoff': configOrCallback?.retries?.backoff,
+                'cloudflare.workflow.retries.delay': configOrCallback?.retries?.delay,
+                'cloudflare.workflow.retries.limit': configOrCallback?.retries?.limit,
+                [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.faas.cloudflare.workflow',
+                [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'task',
+              },
+            },
+            async span => {
+              try {
+                const result = await maybeCallback();
+                span.setStatus({ code: 1 });
+                return result;
+              } catch (error) {
+                captureException(error, { mechanism: { handled: true, type: 'cloudflare' } });
+                throw error;
+              } finally {
+                this._ctx.waitUntil(flush(2000));
+              }
+            },
+          );
+        });
+      });
+    } else {
+      throw new Error(
+        'Invalid arguments for `step.do` method. Expected either (name, callback) or (name, config, callback).',
+      );
+    }
+  }
+
+  public async sleep(name: string, duration: WorkflowSleepDuration): Promise<void> {
+    return this._step.sleep(name, duration);
+  }
+
+  public async sleepUntil(name: string, timestamp: Date | number): Promise<void> {
+    return this._step.sleepUntil(name, timestamp);
+  }
+
+  public async waitForEvent<T extends Rpc.Serializable<T>>(
+    name: string,
+    options: { type: string; timeout?: WorkflowTimeoutDuration | number },
+  ): Promise<WorkflowStepEvent<T>> {
+    return this._step.waitForEvent<T>(name, options);
+  }
+}
+
+/**
+ *
+ * @param optionsCallback
+ * @param WorkFlowClass
+ * @returns
+ */
+export function instrumentWorkflowWithSentry<
```

Comment:
```suggestion
/**
 * Instruments a Cloudflare Workflow class with Sentry.
 * 
 * @example
 * ```typescript
 * const InstrumentedWorkflow = instrumentWorkflowWithSentry(
 *   (env) => ({ dsn: env.SENTRY_DSN }),
 *   MyWorkflowClass
 * );
 * 
 * export default InstrumentedWorkflow;
 * ```
 * 
 * @param optionsCallback - Function that returns Sentry options to initialize Sentry
 * @param WorkflowClass - The workflow class to instrument  
 * @returns Instrumented workflow class with the same interface
 */
export function instrumentWorkflowWithSentry<
```